cupyx/scipy/ndimage/_filters.py

```diff
@@ -356,19 +361,20 @@ def gaussian_filter(input, sigma, order=0, output=None, mode="reflect",
     sigmas = _util._fix_sequence_arg(sigma, input.ndim, 'sigma', float)
     orders = _util._fix_sequence_arg(order, input.ndim, 'order', int)
     truncate = float(truncate)
+    weights_dtype = _util._init_weights_dtype(input)
 
     def get(param):
         sigma, order = param
         radius = int(truncate * float(sigma) + 0.5)
         if radius <= 0:
             return None
-        return _gaussian_kernel1d(sigma, order, radius)
+        return _gaussian_kernel1d(sigma, order, radius, dtype=weights_dtype)
 
     return _run_1d_correlates(input, list(zip(sigmas, orders)), get, output,
                               mode, cval, 0)
 
 
-def _gaussian_kernel1d(sigma, order, radius):
+def _gaussian_kernel1d(sigma, order, radius, dtype=float):
```

Comment:
nit: using numpy dtype
```suggestion
def _gaussian_kernel1d(sigma, order, radius, dtype=cupy.float64):
```